src/vos/evtree.c

```diff
@@ -3395,70 +3395,14 @@ int
 evt_remove_all(daos_handle_t toh, const struct evt_extent *ext,
 	       const daos_epoch_range_t *epr)
 {
-	struct evt_context	*tcx;
-	struct evt_entry	*entry;
-	struct evt_entry_array	 ent_array = { 0 };
-	struct evt_filter	 filter = {0};
-	int			 rc;
-	struct evt_rect		 rect;
-
-	tcx = evt_hdl2tcx(toh);
-	if (tcx == NULL)
-		return -DER_NO_HDL;
-
-	rect.rc_ex = *ext;
-	rect.rc_epc = epr->epr_hi;
-	rect.rc_minor_epc = EVT_MINOR_EPC_MAX;
+	struct evt_entry_in	entry = {0};
 
-	evt_ent_array_init(&ent_array);
+	entry.ei_rect.rc_ex = *ext;
+	entry.ei_bound = entry.ei_rect.rc_epc = epr->epr_hi;
+	entry.ei_rect.rc_minor_epc = EVT_MINOR_EPC_MAX;
```

Comment:
That is good solution, but there is a boundary case that some update via a distributed transaction may have already used the EVT_MINOR_EPC_MAX. So we need to redefine the maximum sub-operations within a distributed transaction to guarantee that EVT_MINOR_EPC_MAX can be reserved for current purpose.

Reply:
I added a new definition for this.